pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/OfflineClusterIntegrationTest.java

```diff
@@ -858,6 +858,28 @@ public void testDefaultColumns()
     _tableSizeAfterRemovingIndex = getTableSize(getTableName());
   }
 
+  @Test
+  public void testDisableGroovyQueryTableConfigOverride()
+      throws Exception {
+    TableConfig tableConfig = getOfflineTableConfig();
+    tableConfig.setQueryConfig(new QueryConfig(5L, Boolean.TRUE));
+    updateTableConfig(tableConfig);
+
+    reloadOfflineTable(getTableName());
```

Comment:
We don't need to reload the table because that won't update the table config on broker, which is ZK watcher based. We can keep sending queries using `TestUtils.waitForCondition()` until the query fails.

Reply:
I added the waitForCondition but for some reason the `testDistinctCountHll` [test](https://github.com/apache/pinot/runs/5211606721?check_suite_focus=true) is flaky when I remove the reload. Any ideas why this is happening?